rotkehlchen/db/updates.py

```diff
@@ -154,29 +156,72 @@ def update_rpc_nodes(self, to_version: int) -> None:
             new_default_nodes=new_default_nodes,
         )
 
+    def update_contracts(self, local_version: int, to_version: int) -> None:
+        """Updates evm contracts and ABIs in globaldb"""
+        for version in range(local_version + 1, to_version + 1):
+            success, updated_contracts = self._get_file_content(
+                file_url=CONTRACTS_URL.format(branch=self.branch, version=version),
+                update_type=UpdateType.CONTRACTS,
+            )
+            if success is False:
+                return
+
+            log.info(f'Applying update for contracts to v{version}')
+            new_abi_data = []
+            for single_abi_data in updated_contracts['abis_data']:
+                new_abi_data.append((
+                    single_abi_data['id'],
+                    single_abi_data['value'],
+                    single_abi_data['name'],
+                ))
+
+            new_contracts_data = []
+            for single_contract_data in updated_contracts['contracts_data']:
+                new_contracts_data.append((
+                    single_contract_data['address'],
+                    single_contract_data['chain_id'],
+                    single_contract_data.get('name'),
+                    single_contract_data['abi'],
+                    single_contract_data.get('deployed_block'),
+                ))
+
+            try:
+                with GlobalDBHandler().conn.write_ctx() as write_cursor:
+                    write_cursor.executemany(
+                        'INSERT OR REPLACE INTO contract_abi(id, value, name) VALUES(?, ?, ?)',
+                        new_abi_data,
+                    )
```

Comment:
This one doesn't need to be in the try/exceptment but I believe is here so it can be rolled back in the same transaction right?

Reply:
Yes